homeassistant/components/tradfri/cover.py

```diff
@@ -21,48 +33,63 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
 class TradfriCover(TradfriBaseDevice, CoverEntity):
     """The platform class required by Home Assistant."""
 
-    def __init__(self, device, api, gateway_id):
+    def __init__(
+        self,
+        device: Command,
+        api: Callable[[Command | list[Command]], Any],
+        gateway_id: str,
+    ) -> None:
         """Initialize a cover."""
         super().__init__(device, api, gateway_id)
         self._attr_unique_id = f"{gateway_id}-{device.id}"
 
         self._refresh(device)
 
     @property
-    def extra_state_attributes(self):
+    def extra_state_attributes(self) -> dict[str, str] | None:
         """Return the state attributes."""
         return {ATTR_MODEL: self._device.device_info.model_number}
 
     @property
-    def current_cover_position(self):
+    def current_cover_position(self) -> int | None:
         """Return current position of cover.
 
         None is unknown, 0 is closed, 100 is fully open.
         """
-        return 100 - self._device_data.current_cover_position
+        if not self._device_data:
+            return None
+        return 100 - cast(int, self._device_data.current_cover_position)
 
-    async def async_set_cover_position(self, **kwargs):
+    async def async_set_cover_position(self, **kwargs: Any) -> None:
         """Move the cover to a specific position."""
+        if not self._device_control:
+            return None
         await self._api(self._device_control.set_state(100 - kwargs[ATTR_POSITION]))
 
-    async def async_open_cover(self, **kwargs):
+    async def async_open_cover(self, **kwargs: Any) -> None:
         """Open the cover."""
+        if not self._device_control:
+            return None
         await self._api(self._device_control.set_state(0))
 
-    async def async_close_cover(self, **kwargs):
+    async def async_close_cover(self, **kwargs: Any) -> None:
         """Close cover."""
+        if not self._device_control:
+            return None
         await self._api(self._device_control.set_state(100))
 
-    async def async_stop_cover(self, **kwargs):
+    async def async_stop_cover(self, **kwargs: Any) -> None:
         """Close cover."""
+        if not self._device_control:
+            return None
```

Comment:
```suggestion
            return
```

Reply:
me too, it is in the works....and due to typing I have identified at least 2 bugs.